modules/apps/portal-security-audit/portal-security-audit-router/src/main/java/com/liferay/portal/security/audit/router/internal/DefaultAuditRouter.java

```diff
@@ -121,54 +119,15 @@ protected void modified(Map<String, Object> properties) {
 		policyOption = ReferencePolicyOption.GREEDY
 	)
 	protected void setAuditMessageProcessor(
-		AuditMessageProcessor auditMessageProcessor,
-		Map<String, Object> properties) {
+		AuditMessageProcessor auditMessageProcessor) {
 
-		String[] eventTypes = _getEventTypes(properties);
-
-		if ((eventTypes.length == 1) && eventTypes[0].equals(StringPool.STAR)) {
-			_globalAuditMessageProcessors.add(auditMessageProcessor);
-
-			return;
-		}
-
-		for (String eventType : eventTypes) {
-			Set<AuditMessageProcessor> auditMessageProcessorsSet =
-				_auditMessageProcessors.get(eventType);
-
-			if (auditMessageProcessorsSet == null) {
-				auditMessageProcessorsSet = new HashSet<>();
-
-				_auditMessageProcessors.put(
-					eventType, auditMessageProcessorsSet);
-			}
-
-			auditMessageProcessorsSet.add(auditMessageProcessor);
-		}
+		_globalAuditMessageProcessors.add(auditMessageProcessor);
```

Comment:
@W2jx Here `*` means a wildcard match, so the logic adds the one with `*` to the `global` collection. If you want to eliminate the need to divide the processor into global & non-global, you need to first investigate where and how `eventType` is used. Do they have any logic to get with any specific event type? You need to update that part too.